src/pubtools/_marketplacesvm/tasks/community_push/command.py

```diff
@@ -443,58 +442,68 @@ def _check_product_in_rhsm(self, enriched_push_items: List[EnrichedPushItem]) ->
 
         Args:
-            enriched_push_item
-                Dictionary with the storage account name and the push items.
+            push_queue
+                Iterator with the required data to upload.
         Returns:
             Dictionary with the resulting operation for the Collector service.
         """
-        result = []
-        for storage_account, push_items_and_sa in enriched_push_item.items():
-            # Setup the threading
-            to_await = []
-            out_pi = []
-            push_items = push_items_and_sa.push_items
-            executor = Executors.thread_pool(
-                name="pubtools-marketplacesvm-community-push-regions",
-                max_workers=min(max(len(push_items), 1), self._PROCESS_THREADS),
-            )
+        to_await = []
+        upload_result = []
+        executor = Executors.thread_pool(
+            name="pubtools-marketplacesvm-community-push",
+            max_workers=self._REQUEST_THREADS,
+        )
 
-            # Prepare the sharing accounts
-            sharing_accts: SharingAccounts = push_items_and_sa.sharing_accounts
-            additional_args = {}
-            extra_args = ["accounts", "sharing_accounts", "snapshot_accounts"]
-            for arg in extra_args:
-                content = sharing_accts.get(arg)
-                if content:
-                    additional_args[arg] = content
-
-            # Upload the push items in parallel
-            log.info("Uploading to the storage account %s", storage_account)
-            for pi in push_items:
-                to_await.append(
-                    executor.submit(self._upload, storage_account, pi, **additional_args)
-                )
+        # consume the queue
+        for data in push_queue:
+            to_await.append(executor.submit(self._upload, **data))
 
-            # Wait for all results
-            for f_out in to_await:
-                out_pi.append(f_out.result())
-
-            # Append the data for collection
-            for pi in out_pi:
-                result.append(
-                    {
-                        "push_item": pi,
-                        "state": pi.state,
-                        "image_id": pi.image_id,
-                        "image_name": name_from_push_item(pi),
-                    }
-                )
-        return result
+        # wait for results
+        for f_out in to_await:
+            upload_result.append(f_out.result())
+
+        # Return the data for collection
+        return [
+            {
+                "push_item": pi,
+                "state": pi.state,
+                "image_id": pi.image_id,
+                "image_name": name_from_push_item(pi),
+            }
+            for pi in upload_result
+        ]
+
+    def _data_to_upload(
+        self, enriched_push_items: List[EnrichedPushItem]
+    ) -> Iterator[Dict[str, Any]]:
```

Comment:
Could we return `TypedDict` instead of `Dict[str, Any]` ?

Reply:
Sure, I'll adjust it

Reply:
Now it's adjusted to use `TypedDict`, PTAL again